services/appflowy-collaborate/src/collab/storage.rs

```diff
@@ -270,23 +283,23 @@ where
     Ok(())
   }
 
-  async fn insert_new_collab(
+  async fn batch_insert_new_collab(
     &self,
     workspace_id: &str,
     uid: &i64,
-    params: CollabParams,
+    params_list: Vec<CollabParams>,
   ) -> AppResult<()> {
-    params.validate()?;
-
     self
       .check_write_workspace_permission(workspace_id, uid)
       .await?;
+    for params in &params_list {
+      self
+        .access_control
+        .update_policy(uid, &params.object_id, AFAccessLevel::FullAccess)
```

Comment:
If `batch_insert_collabs` should work on batches, should that for loop be a batch as well?

Reply:
yes. it should be. Because shuheng is working on access control. So I don't want to modify the access control. I will leave a todo for myself